Expose the series property of Highchart

peterchibunna opened the pull request:
Exposing the `series` property to the HighCharts object allows for more complex charts to be manipulated like the original highcharts

avdaredevil commented:
Hey @peterchibunna, could you give me an example.
The way I handled those cases were by supporting the manipulation of series objects via functions like `setData`, `addData`, `pushData`.

And to modify the series themselves using
- `addSeries`: Let's you add a new series with multiple options
- `updateSeries`: Let's you update batch options (`{}`,`series_index`) or individual (`key`,`value`,`series_index`)
- `removeSeries`: Remove @ index.

I can see the need for getting a series by index, and that is done via an internal method called `_getSeries(z)`. If you want I can expose that?

And all other attributed of the chart can be fetched using the public `_chart` property (eg. `highchart._chart.series` or `highchart._chart.xAxis`)

avdaredevil commented:
Since i have not recieved a reply from you yet, I'm going to assume the answer above addressed your issues and will close this pull request.

peterchibunna commented:
No I actually sent a reply from my phone but I'm surprised it's not here. Let me go copy and paste it here.

avdaredevil commented:
Your proposition isn't as much dangerous, as it is redundant. There is another issue: that if the chart object is directly modified with a new Series array, the data binding would fail, as the Memory address for the new series object would not reflect the `highcharts-chart` property `series`.

In regards to your concern: Only the `_chart` property is _read-only_ __not__ it's sub properties. Thus, you can easily go ahead and run `highchart._chart.series = []`. This is to prevent the user from accidentally overwriting the backing chart object, from/on which all properties and methods rely on.

If you're still having issues, paste an example here or on StackOverflow, and I can answer it for you. I don't see a way to mantain safe binding by exposing the series object directly.

However, If I can see a compelling enough reason to add this property, I'll accept the pull. Fair?

avdaredevil commented:
Hey @peterchibunna if you could acknowledge that you received this message. I can go ahead tag and tag it accordingly.

peterchibunna commented:
You can do as you deem fit. I'll experiment further.

peterchibunna commented:
If I call `app.$.temperatureChart._chart.series[0].addPoint([(new Date()).getTime(), event.detail.temperature]);` from my app, the chart disappears.

However if I paste the code in the browser console, it works fine.

What could be causing my issue?

avdaredevil commented:
You might be better off using `app.$.temperatureChart.addData((new Date()).getTime(), event.detail.temperature);`

The only reason I can see that happening is if you run this before your WebComponents have loaded [ie. not running it after [WebComponentsReady Event](http://webcomponents.org/polyfills/html-imports/#the-webcomponentsready-event)] and that verifying whether `event.detail.temperature` has the correct data.

peterchibunna commented:
It adds the data correctly but the chart dots and paintings are lost by the time I navigate to the chart. The data is there. Could you please add a simpler example on your demo page to initialize chart with data like
`<highcharts-chart vs-time id="cvDilation" type="spline" title="Cervical Dilation" data='[[1442573315000,36],[1442917213000,36.6]]'></highcharts-chart>`

Sorry, I'm still finding it uneasy to use your plugin, But I have to.

avdaredevil commented:
I'm sorry you're having issues using it. I tried to make it as intuitive as possible. I've put it up right [here](https://avdaredevil.github.io/highcharts-chart/Demo2.html).

It might be easier to help you over stack-overflow where I could see your code.
However, I've had a similar issue with pages with multiple tabs, that the charts were being re-sized to a certain size when I went to that page. One way to fix such a scenario is to set a size on the `<highcharts-chart>` container, and then run `app.$.cvDilation.resizeChart()` when you go to the page where the chart resides.

Does that help?

peterchibunna commented:
Thanks very much. I had already figured out the issue in my code. I guess my javascript skills are not yet good. The first issue I reported was being caused by the data i'm passing not being a number. So just converting it using the `Number` function rectified it. Secondly, where I assigned data to the highcharts-chart was being misinterpreted by Polymer for data binding. I just changed the code a little to 

`<highcharts-chart vs-time id="cvDilation" type="spline" title="Cervical Dilation" data='[ [1442573315000,36],[1442917213000,36.6] ]'></highcharts-chart>`

Added space before the data array itself and it worked.

peterchibunna commented:
Can you do me another demo for a chart with two series: I'm plotting blood pressure data: `systolic` series and `diastolic` series: by simple assignment to the data as in above, as well as using the `addSeries` function.  Thanks very much for your time.

avdaredevil commented:
So the library only allows simplistic binding with the Direct Assign/Bind method. If you have multiple series. You simply have to script it like [so](https://avdaredevil.github.io/highcharts-chart/Demo.html).

No Probs :D, you're making me to quite a bit of work. I'm hoping that this might also help other people.
